src/wallet.js

```diff
@@ -439,12 +440,62 @@ export class Wallet extends React.Component {
 
     this.setState({
       requesterOrigin: origin,
-      requestPending: true,
       requestedAmount: `${params.amount || ''}`,
       requestedPublicKey: params.pubkey,
     });
   }
 
+  onSendCustomTransactionRequest(params, origin) {
+    if (!params.format || !params.network || !params.transaction) {
+      if (!params.network) this.addError(`Request did not specify a network`);
+      if (!params.format) this.addError(`Request did not specify a transaction format`);
+      if (!params.transaction) this.addError(`Request did not specify a transaction`);
+      return;
+    }
+
+    let requestedNetwork;
+    try {
+      requestedNetwork = new URL(params.network).origin;
+    } catch (err) {
+      this.addError(`Request network is invalid: "${params.network}"`);
+      return;
+    }
+
+    const walletNetwork = new URL(this.props.store.networkEntryPoint).origin;
+    if (requestedNetwork !== walletNetwork) {
+      this.props.store.setNetworkEntryPoint(requestedNetwork);
+      this.addWarning(
+        `Changed wallet network from "${walletNetwork}" to "${requestedNetwork}"`,
+      );
+    }
+
+    const transaction = new web3.Transaction();
+    const inputs = JSON.parse(params.transaction);
+
+    inputs.map(input  => {
```

Comment:
Yes, I also wanted to do this, but SDK does not serialize an unsigned transaction, can you give an example how to do it?

`
public serialize(): Buffersource
Serialize the Transaction in the wire format.
The Transaction must have a valid signature before invoking this method

I have error
Uncaught Error: Transaction recentBlockhash required
`

Reply:
Check out https://github.com/solana-labs/solana-web3.js/commit/85b354e0db8e6bc30328df45c3c55816d3489ee6

Reply:
need to fill out `recentBlockhash`, what if the user did not immediately confirm the country transaction, it needs to be filled again on the wallet side?

Reply:
Setting a placeholder is fine because the wallet will always set the `recentBlockhash` when signing. Something like:

```
transaction.recentBlockhash = new PublicKey(0).toBase58();
```